stan/math/prim/prob/ordered_probit_lpmf.hpp

```diff
@@ -84,12 +86,13 @@ return_type_t<T_loc, T_cut> ordered_probit_lpmf(const T_y& y,
     int K = c_vec[i].size() + 1;
 
     if (y_vec[i] == 1) {
-      logp_n += log1m(Phi(lambda_vec[i] - c_vec[i].coeff(0)));
+      logp_n += std_normal_lcdf(c_vec[i].coeff(0) - lambda_vec[i]);
     } else if (y_vec[i] == K) {
-      logp_n += log(Phi(lambda_vec[i] - c_vec[i].coeff(K - 2)));
+      logp_n += std_normal_lcdf(lambda_vec[i] - c_vec[i].coeff(K - 2));
     } else {
-      logp_n += log(Phi(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 2))
-                    - Phi(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 1)));
+      logp_n += log_diff_exp(
+          std_normal_lcdf(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 2)),
+          std_normal_lcdf(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 1)));
```

Comment:
I've been thinking more about this because there's an alternative way to write this
```cpp
      logp_n += log_diff_exp(
          std_normal_lccdf(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 1)),
          std_normal_lccdf(lambda_vec[i] - c_vec[i].coeff(y_vec[i] - 2)));
```
but it's not obvious which is more numerically stable.
I don't think it makes much difference when lambda is between the cutpoints but when both cutpoints are lower than lambda the CCDF version is better (and when lambda is below both cutpoints the CDF version is better)
(And of course the code must use `*_cdf(-x)` instead of `*_ccdf(x)`)

Reply:
I believe the ```std_normal_lcdf``` will be more stable here since the ```lccdf``` does not have the additional numerical approximations for boundary values

Reply:
Well, yes. I wrote `lccdf` because that's how I was thinking about it when I derived the equality but you'd want to change that to a negated-argument `lcdf`.
```cpp
      logp_n += log_diff_exp(
          std_normal_lcdf(c_vec[i].coeff(y_vec[i] - 1) - lambda_vec[i]),
          std_normal_lcdf(c_vec[i].coeff(y_vec[i] - 2) - lambda_vec[i]));
```

Reply:
Oh right, missed that sorry! Will push those changes now